march_joint_inertia_controller/include/march_joint_inertia_controller/joint_trajectory_inertia_controller.h

```diff
@@ -0,0 +1,211 @@
+
+namespace joint_trajectory_controller
+{
+typedef trajectory_interface::QuinticSplineSegment<double> SegmentImpl;
+typedef JointTrajectorySegment<SegmentImpl> Segment;
+typedef typename Segment::State State;
+}  // namespace joint_trajectory_controller
+
+template <>
+class HardwareInterfaceAdapter<hardware_interface::EffortJointInterface, joint_trajectory_controller::State>
+{
+public:
+  HardwareInterfaceAdapter() : joint_handles_ptr_(nullptr)
+  {
+  }
+
+  /**
+   * \brief Initialize the controller by establishing the pointer to the joints, the pid calculators and the inertia
+   * estimators
+   */
+  bool init(std::vector<hardware_interface::JointHandle>& joint_handles, ros::NodeHandle& nh)
+  {
+    joint_handles_ptr_ = &joint_handles;
+    const unsigned int num_joints_ = joint_handles_ptr_->size();
+
+    // Initialize PIDs
+    pids_.resize(num_joints_);
+    for (unsigned int i = 0; i < pids_.size(); ++i)
+    {
+      ros::NodeHandle joint_nh(nh, std::string("gains/") + joint_handles[i].getName());
+
+      // Init PID gains from ROS parameter server
+      pids_[i].reset(new control_toolbox::Pid());
+      if (!pids_[i]->init(joint_nh))
+      {
+        ROS_WARN_STREAM("Failed to initialize PID gains from ROS parameter server.");
+        return false;
+      }
+    }
+
+    // Obtain inertia estimator parameters from server
+    double lambda[2];
+    int alpha_filter_size[2];
+    ros::NodeHandle rotary_estimator_nh(nh, std::string("inertia_estimator/rotary"));
+    rotary_estimator_nh.param("std_samples", samples_, 100);
+    rotary_estimator_nh.param("lambda", lambda[0], 0.96);
+    rotary_estimator_nh.param("alpha_filter_size", alpha_filter_size[0], 12);
+
+    ros::NodeHandle linear_estimator_nh(nh, std::string("inertia_estimator/linear"));
+    linear_estimator_nh.param("lambda", lambda[1], 0.96);
+    linear_estimator_nh.param("alpha_filter_size", alpha_filter_size[1], 12);
+
+    // Initialize the estimator parameters
+    inertia_estimators_.resize(num_joints_);
+    for (unsigned int j = 0; j < num_joints_; ++j)
+    {
+      inertia_estimators_[j].setNodeHandle(nh);
+      inertia_estimators_[j].setLambda(lambda[(int)floor(j / 2) % 2]);  // Produce sequence 00110011
+      inertia_estimators_[j].setAcc_size(alpha_filter_size[(int)floor(j / 2) % 2]);
+      inertia_estimators_[j].configurePublisher(joint_handles[j].getName());
+    }
+    return true;
+  }
+
+  /**
+   * \brief Updates the commanded effort for each individual joint and estimates the inertia on each joint and publishes
+   * that information on a topic
+   */
+  void updateCommand(const ros::Time& /*time*/, const ros::Duration& period,
+                     const joint_trajectory_controller::State& /*desired state*/,
+                     const joint_trajectory_controller::State& state_error)
+  {
+    const unsigned int num_joints_ = joint_handles_ptr_->size();
+
+    // Preconditions
+    if (!joint_handles_ptr_)
+      return;
```

Comment:
We like to use brackets here
```suggestion
    if (!joint_handles_ptr_)
    {
      return;
    }
```